app/domain/errors.rb

```diff
@@ -273,6 +273,77 @@ module AuthnK8s
           msg:  "Invalid Kubernetes host id: {0}. Must end with <namespace>/<resource_type>/<resource_id>",
           code: "CONJ00048E"
         )
+
+        CSRMissingCNEntry = ::Util::TrackableErrorClass.new(
+          msg:  "CSR [subject: '{0-subject}', spiffe_id: '{1-spiffe-id}'] must have a CN (common name) entry.",
+          code: "CONJ00058E"
+        )
+
+        CertMissingCNEntry = ::Util::TrackableErrorClass.new(
+          msg:  "Cert [subject: '{0-subject}', san: '{1-san}'] must have a CN (common name) entry.",
+          code: "CONJ00059E"
+        )
+
+        PodNameMismatch = ::Util::TrackableErrorClass.new(
+            msg:  "Pod: {0-pod-name}, resource name does not match: {1-self-name}.",
+            code: "CONJ00060E"
+        )
+
+
+        PodStatefulSetNameMismatch = ::Util::TrackableErrorClass.new(
+            msg:  "Pod: {0-pod-name}, StatefulSet metadata name: {1-stateful-set-metadata-name} does not match: " \
+                  "{2-stateful-set-name}.",
+            code: "CONJ00061E"
+        )
+
+        PodDoesNotBelongToAStatefulSet = ::Util::TrackableErrorClass.new(
```

Comment:
It seems that the following errors can be merged into one and have another input:
- `PodDoesNotBelongToAStatefulSet`
- `PodDoesNotBelongToAReplicaSet`
- `PodDoesNotBelongToADeploymentConfig`
- `PodDoesNotBelongToAReplicationController`
- `PodDoesNotBelongToADeployment `